tests/tests.h

```diff
@@ -591,6 +592,42 @@ struct MPILogInitAll
 };
 
 
+#ifdef DEAL_II_COMPILER_CUDA_AWARE
+// By default, all the ranks will try to access the device 0. We can distribute
+// the load better by using a random device which is done in this function. In
+// case tests use MPI, we make sure to set subsequent devices. MPI needs to be
+// initialized before using this function.
+// Also initialize a dummy handle that makes sure that unused memory is released
+// before the device shuts down.
+void
+init_cuda(bool use_mpi = false)
+{
+  static Utilities::CUDA::Handle cuda_handle;
+#  ifndef DEAL_II_WITH_MPI
+  Assert(use_mpi == false, ExcInternalError());
+#  endif
+  const unsigned int my_id =
+    use_mpi ? Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) : 0;
```

Comment:
do you really need `?:` here? Shouldn't the function just return zero if no MPI is used?

Reply:
The function returns zero if we don't compile with `MPI` support. 

If we say that the test doesn't require `MPI`, `MPI` is normally not initialized and calling `Utilities::MPI::this_mpi_process()` is not allowed.

Reply:
Oh, I see -- the second part is the important part: What to do when running a single-process job when all of deal.II has been compiled with MPI. Thanks for the clarification.